graphql-jpa-query-schema/src/test/java/com/introproventures/graphql/jpa/query/schema/StarwarsSchemaBuildWithDistinctTest.java

```diff
@@ -0,0 +1,83 @@
+package com.introproventures.graphql.jpa.query.schema;
+
+import com.introproventures.graphql.jpa.query.schema.impl.GraphQLJpaExecutor;
+import com.introproventures.graphql.jpa.query.schema.impl.GraphQLJpaSchemaBuilder;
+import graphql.schema.GraphQLSchema;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.persistence.EntityManager;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+        webEnvironment= SpringBootTest.WebEnvironment.NONE
+)
+public class StarwarsSchemaBuildWithDistinctTest {
+    @SpringBootApplication
+    static class TestConfiguration {
+        @Bean
+        public GraphQLJpaSchemaBuilder graphQLSchemaBuilder(EntityManager entityManager) {
+            return new GraphQLJpaSchemaBuilder(entityManager)
+                    .name("Starwars")
+                    .description("Starwars Universe Schema")
+                    .useDistinctParameter(true);
+        }
+
+        @Bean
+        public GraphQLExecutor graphQLExecutor(final GraphQLSchemaBuilder graphQLSchemaBuilder) {
+            return new GraphQLJpaExecutor(graphQLSchemaBuilder.build());
+        }
+    }
+
+    @Autowired
+    private GraphQLJpaSchemaBuilder builder;
+
+    @Autowired
+    private GraphQLExecutor executor;
+
+    @Before
+    public void setup() {
+    }
+
+    @Test
+    public void correctlyDerivesPageableSchemaFromGivenEntities() {
+        //when
+        GraphQLSchema schema = builder.build();
+
+        // then
+        assertThat(schema)
+                .describedAs("Ensure the schema is generated")
+                .isNotNull();
+
+
+        //then
+        assertThat(schema.getQueryType()
+                .getFieldDefinition("CodeLists").getArguments()
+        )
+                .describedAs("Ensure query has three arguments")
+                .hasSize(3);
+    }
+
+    @Test
+    public void distinctFalse() {
+        //given
+        String query = "{ Books(distinct: false) { select { genre } }}";
```

Comment:
@miklemv I think it makes more sense to use `distinct` inside select field, i.e.

`{ Books { select( distinct: false ) { genre } }}`